docs/GLOSSARY.md

```diff
@@ -0,0 +1,98 @@
+# Glossary
+Snap is simple in scope and it becomes more simple when you know the terminology we use throughout the project. Here they are:
+
+* [Glossary](#glossary)
+    * [Config: Global Config](#config-global-config)
+    * [Config: Global Options](#config-global-options)
+    * [Config: Metric Config](#config-metric-config)
+    * [Config: Plugin Config](#config-plugin-config)
+    * [Metric Catalog](#metric-catalog)
+    * [Metric: Dynamic](#metric-dynamic)
+    * [Metric: Namespace](#metric-namespace)
+    * [Metric Namespace: Dynamic Element](#metric-namespace-dynamic-element)
+    * [Metric Namespace: Static Element](#metric-namespace-static-element)
+    * [Plugin](#plugin)
+    * [Plugin Type: Collector](#plugin-type-collector)
+    * [Plugin Type: Processor](#plugin-type-processor)
+    * [Plugin Type: Publisher](#plugin-type-publisher)
+    * [Snap](#snap)
+    * [Snap Telemetry](#snap-telemetry)
+    * [snapctl](#snapctl)
+    * [snapd](#snapd)
+    * [Task](#task)
+    * [Task Manifest](#task-manifest)
+    * [Tribe](#tribe)
+    * [Workflow](#workflow)
+    * [Workflow: Distributed](#workflow-distrubted)
+    * [Workflow Manifest](#workflow-manifest) 
+
+### Config: Global Config 
+* Values loaded at runtime of the daemon ([reference](https://github.com/intelsdi-x/snap/blob/master/docs/SNAPD_CONFIGURATION.md))
```

Comment:
I really like the idea to link to an actual context! 👍 

Can you use a relative path for url so links work from other branch/fork?